app/react/src/client/preview/render.tsx

```diff
@@ -28,9 +37,50 @@ export const render: ArgsStoryFn<ReactFramework> = (args, context) => {
 
 const renderElement = async (node: ReactElement, el: Element) =>
   new Promise((resolve) => {
-    ReactDOM.render(node, el, () => resolve(null));
+    // Create Root Element conditionally for new React 18 Root Api
+    const root = getReactRoot(el);
+
+    if (root) {
+      root.render(node);
+      setTimeout(() => {
```

Comment:
Is `root.render(node)` async?

Reply:
Please take a look at https://github.com/reactwg/react-18/discussions/5 (section "What about the render callback?").

And take a look at the following sub-thread. This will explain the three available methods of alternatives to the render callback in detail https://github.com/reactwg/react-18/discussions/5#discussioncomment-796012

Reply:
A part from the sub-thread: „Use the ref callback if you want to wait until the root component commits, use requestIdleTimeout if you want to just wait for low-priority idle time, and use setTimeout if you just want to wait until after React yields the first time.“

Reply:
> Is `root.render(node)` async?

„ In the legacy root API, you could pass render a callback that is called after the component is rendered or updated.

In the New Root API, we’ve removed this callback.

With partial hydration and progressive SSR, the timing for this callback would not match what the user expects. To avoid the confusion moving forward, we recommend using requestIdleCallback, setTimeout, or a ref callback on the root.“ - https://github.com/reactwg/react-18/discussions/5

Reply:
@shilman Can I somehow support further? :)

Reply:
Thanks @valentinpalkovic -- I'm just waiting for the thumbs up from @chantastic who's going to do some testing of this. I definitely want to get this out as part of 6.5!